runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -79,6 +81,23 @@ public Integer getPort() {
     public ClientMsgHandler msgHandler = new ClientMsgHandler(this)
             .generateHandlers(MethodHandles.lookup(), this);
 
+    /** Handles the LOGUNIT_METRICS_RESPONSE
+     *
+     * @param msg   Incoming message
+     * @param ctx   Context
+     * @param r     Router
+     * @return      byte[] of the LogUnitMetrics object.
+     */
+    @ClientHandler(type=CorfuMsgType.LOGUNIT_METRICS_RESPONSE)
+    private static Object handleMetricsResponse(CorfuPayloadMsg<byte[]> msg,
+                                                ChannelHandlerContext ctx, IClientRouter r) {
+        try {
+            return LogUnitMetrics.parseFrom(msg.getPayload());
+        } catch (InvalidProtocolBufferException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)